webviz.html

```diff
@@ -460,18 +493,71 @@ <h1>UT AUTOmata Web Interface</h1>
       xhr.send();
     }
 
+    function comparePoses(pose1, pose2) {
+      const [x1, y1, theta1] = pose1;
+      const [x2, y2, theta2] = pose2;
+      const positionThreshold = 0.1;
+      const orientationThreshold = 0.05;
+
+      return (
+        Math.abs(x1 - x2) < positionThreshold &&
+        Math.abs(y1 - y2) < positionThreshold &&
+        Math.abs(theta1 - theta2) < orientationThreshold
+      );
+    }
+
     function setLocationButton() {
       setLocation = true;
       setNavGoal = false;
+      resetNavGoals = false;
+      setWaypoints = false;
+      document.getElementById("resetNav").style = "";
       document.getElementById("setLocation").style = "border-style:inset;";
     }
 
     function setNavGoalButton() {
       setLocation = false;
       setNavGoal = true;
+      resetNavGoals = false;
+      setWaypoints = false;
+      document.getElementById("resetNav").style = "";
       document.getElementById("setNavGoal").style = "border-style:inset;";
     }
 
+    function setWaypointsButton() {
+      if (!setWaypoints) {
+        setLocation = false;
+        setNavGoal = false;
+        resetNavGoals = false;
+        setWaypoints = true;
+        document.getElementById("resetNav").style = "";
+        document.getElementById("setWaypoints").style = "border-style:inset;";
+      } else {
+        setWaypoints = false;
+        document.getElementById("setWaypoints").style = "";
+        robot_interface.gotoWaypoints();
+      }
+    }
+
+    function resetNavButton() {
+      if (!resetNavGoals) {
+        setNavGoal = false;
+        setLocation = false;
+        resetNavGoals = true;
+        setWaypoints = false;
+        mouseDown = false;
+        robot_interface.waypoints = [];
+        document.getElementById("setLocation").style = "";
+        document.getElementById("setNavGoal").style = "";
+        document.getElementById("setWaypoints").style = "";
+        robot_interface.setNavGoal(1111, 1111, 1111);  // all 1111 -> indicates reset of nav goals
```

Comment:
Don't use hardcoded coordinates. This is a good use case for adding a new JSON field to the websocket callback.